front/src/common/BootstrapSNCF/MultiSelectSNCF.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+type MultiSelectSNCFProps = {
+  multiSelectTitle: string;
+  multiSelectSubTitle?: string;
+  selectOptions: { label: string }[];
+  onChange: React.Dispatch<React.SetStateAction<string[]>>;
+  selectedValues: string[];
+  isSelectAll?: boolean;
+};
+
+const MultiSelectSNCF = ({
+  multiSelectTitle,
+  multiSelectSubTitle,
+  selectOptions,
+  onChange,
+  selectedValues,
+  isSelectAll,
+}: MultiSelectSNCFProps) => {
+  const [multiSelectToggle, setMultiselectToggle] = useState<boolean>(false);
+  const [selectAll, setSelectAll] = useState<boolean>(false);
+
+  const renderOptions = useCallback(
+    (options: { label: string }[]) =>
+      options.map((selectOption, index) => (
+        <option
+          data-id={index}
+          selected={selectedValues.includes(selectOption.label)}
+          key={`selectOption-${selectOption.label}`}
+          value={selectOption.label}
+        >
+          {selectOption.label}
+        </option>
+      )),
+    [selectOptions]
+  );
+
+  const renderSelectToggles = (options: { label: string }[]) =>
+    options.map((toggle, index) => {
+      const { label } = toggle;
+      const isChecked = selectedValues.includes(label);
+      const onClickToggle = () => {
+        onChange(
+          isChecked
+            ? [...selectedValues.filter((selectValue) => selectValue !== label)]
+            : (prevValues) => [...prevValues, label]
+        );
+      };
+      return (
+        <div className="select-menu-item" role="listitem" key={`selectToggle-${label}`}>
+          <div className="custom-control custom-checkbox">
+            <button
+              type="button"
+              data-role="value"
+              data-target={index}
+              role="checkbox"
+              aria-checked="false"
+              className={`custom-control-label w-100 text-left font-weight-medium ${
+                isChecked ? 'active' : ''
+              }`}
+              onClick={onClickToggle}
+            >
+              {label}
+            </button>
+          </div>
+        </div>
+      );
+    });
+
+  const memoizedAllOptionsLabel = useMemo(
+    () => selectOptions.map((option) => option.label),
+    [selectOptions]
+  );
+
+  useEffect(() => {
+    const isAllSelected = memoizedAllOptionsLabel.every((label) => selectedValues.includes(label));
+    if (isAllSelected && !selectAll) setSelectAll(true);
+    if (!isAllSelected && selectAll) setSelectAll(false);
+  }, [selectedValues]);
+
+  useEffect(() => {
+    const isAllSelected = memoizedAllOptionsLabel.every((label) => selectedValues.includes(label));
+    const selects = isAllSelected ? [] : selectedValues;
+    onChange(selectAll ? [...memoizedAllOptionsLabel] : selects);
+  }, [selectAll]);
```

Comment:
```suggestion
  const handleSelectAll = (newSelectAll: boolean) => {
    if (newSelectAll) {
      onChange(options);
    } else {
      onChange([]);
    }
  };
```
We should update selectAll only with the useEffect above, and use this function in the selectAll button
```
    onClick={() => handleSelectAll(!selectAll)}
```

Reply:
Done